addons/account_facturx/models/account_move.py

```diff
@@ -284,20 +284,38 @@ def message_post(self, **kwargs):
             self.env.context.get('default_type', self.type) in self.env['account.move'].get_invoice_types(include_receipts=True)
             or self.env['account.journal'].browse(self.env.context.get('default_journal_id')).type in ('sale', 'purchase')
         ):
-            for attachment in self.env['ir.attachment'].browse(kwargs.get('attachment_ids', [])):
+            # Get attachments.
+            # - 'attachments' is a namedtuple defined in mail.thread looking like:
+            # _Attachment = namedtuple('Attachment', ('fname', 'content', 'info'))
+            # - 'attachment_ids' is a list of ir.attachment records ids.
+            attachments = kwargs.get('attachments', [])
+            if kwargs.get('attachment_ids'):
+                attachments += self.env['ir.attachment'].browse(kwargs['attachment_ids'])
```

Comment:
Seems like we are still not managing all the possible cases:
https://github.com/odoo/odoo/blob/99531f5dd8966fc4ad89d1e9c2886ad701527f3a/addons/mail/models/mail_thread.py#L1818-L1823

I think it would be better to use `_message_post_process_attachments` to be DRY consistent

Reply:
@william-andre Done :+1: 
I removed also `message_post` since not used anymore, is it right?